File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        String sb=new StringBuilder().append(Math.abs((long)x)).reverse().toString();
        try
        {
            return (x>0)?
            Integer.parseInt(sb)
            :
            Integer.parseInt(sb)*-1;
        }
        catch(NumberFormatException e)
        {
                return 0;
        }
    }
}

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        rev=0
        if x>0:
            s=1
        else:
            s=-1
        x=abs(x)
        while x>0:
            dig=x%10
            rev=rev*10+dig
            x=x//10
        rev*=s
        if rev> 2**31 -1 or rev< -2**31 :
            return 0
        return rev